ip/event_manager.go

```diff
@@ -175,28 +186,34 @@ func (m *manager) singleEtcdRun(ctx context.Context) {
 }
 
 func (m *manager) healthChecker(ctx context.Context) {
-	log := logger.Get(ctx)
 	for {
 		healthy := m.checker.IsHealthy(ctx)
 
 		// The eventManager will close the chan when we receive the Stop order and we do not want to send things on a close channel.
 		// Since the checker can take up to 5s to run his checks, this check must be done between the health check and sending the results.
-		if m.isStopping() {
+		if m.isStopped() {
 			return
 		}
 
-		if healthy {
-			m.failingCount = 0
-			m.sendEvent(HealthCheckSuccessEvent)
-		} else {
-			m.failingCount++
-			log.WithField("failing_count", m.failingCount).Error("Node failing")
-			if m.failingCount >= m.config.FailCountBeforeFailover {
-				log.Error("Too many failure, setting status to failing")
-				m.sendEvent(HealthCheckFailEvent)
-			}
+		if !m.isStopping() {
```

Comment:
If manager is in stopping state, we'll do the loop one more time. Is that right? If I understand everything we only leave the health checker once the manager is stopped. That's the core change of this PR right?

Reply:
We'll continue looping (not necessary just one more time, it could be many more) if the IP is stopping, but we wont send any events (since we do not want to interract with the state machine state).

But the main change here is that this method will only stop if the manager is STOPPED and not if it is stopping.